crates/jsonschema/src/keywords/dependencies.rs

```diff
@@ -48,10 +50,10 @@ impl DependenciesValidator {
 }
 
 impl Validate for DependenciesValidator {
-    fn is_valid(&self, instance: &Value, ctx: &mut ValidationContext) -> bool {
+    fn is_valid(&self, instance: &Value, ctx: &mut LightweightContext) -> bool {
```

Comment:
Maybe `LightweightContext` is not needed at all - we'll create the default one in the beginning and then in `is_valid` just don't track anything...